### ISSUE TITLE
equals() generation is broken with a field named other

### ISSUE THREAD

**Issue by anthonyvdotbe** [redhat-developer/vscode-java#1345]
If a class contains a field named `other`, the generated `equals()` method will be broken

##### Environment
- Operating System: Windows 10
- JDK version: 13.0.1
- Visual Studio Code version: 1.43.2
- Java extension version: 0.58.0

##### Steps To Reproduce
1. do `Source Action...` -> `Generate hashCode() and equals()...` in the class below

````java
class Issues {
    private String other;
}
````

##### Current Result
````java
return Objects.equals(other, other.other);
````

##### Expected Result
````java
return Objects.equals(this.other, other.other);
````

**Comment by fbricon** [comment]
Indeed, this occurs when "java.codeGeneration.hashCodeEquals.useJava7Objects": true, this is a bug in upstream Eclipse JDT

**Comment by fbricon** [comment]
I opened https://bugs.eclipse.org/bugs/show_bug.cgi?id=561517

**Comment by fbricon** [comment]
This was fixed 5 years ago
